File: src/models/rushing.player.ts
import Player from './player';

export default class RushingPlayer extends Player {
  rushAttempts!: number;
  yards!: number;
  rushTouchdowns!: number;
  firstDowns!: number;
  longestRush!: number;
  fumbles!: number;

  public convert(json: any): void {
    super.convert(json);
    this.rushAttempts = json.rush_att || 0;
    this.yards = json.rush_yds || 0;
    this.firstDowns = json.rush_first_down || 0;
    this.longestRush = json.rush_long || 0;
    this.fumbles = json.fumbles || 0;
  }
}

File: c-sharp/src/models/RushingPlayer.cs
namespace MaddenImporter
{
    public class RushingPlayer : Player
    {
        public int RushAttempts { get; set; }
        public int Yards { get; set; }
        public int RushTouchdowns { get; set; }
        public int FirstDowns { get; set; }
        public int LongestRush { get; set; }
        public int Fumbles { get; set; }
    }
}
